p2p/switch.go

```diff
@@ -561,14 +563,17 @@ func (sw *Switch) addPeer(pc peerConn) error {
 	return nil
 }
 
-func (sw *Switch) startInitPeer(peer *peer) {
+func (sw *Switch) startInitPeer(peer *peer) error {
 	err := peer.Start() // spawn send/recv routines
 	if err != nil {
 		// Should never happen
 		sw.Logger.Error("Error starting peer", "peer", peer, "err", err)
+		return err
```

Comment:
Oops, this def looks like an oversight, we should have had this return statement even without the error.